test/e2e/atlas/streams_test.go

```diff
@@ -0,0 +1,279 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//go:build e2e || (atlas && debug)
+
+package atlas_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/test/e2e"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20230201004/admin"
+)
+
+const (
+	defaultInstanceName = "DefaultInstance"
+	instanceName        = "E2EInstance"
+	connectionName      = "E2EConnection"
+)
+
+func TestStreams(t *testing.T) {
+	cliPath, err := e2e.AtlasCLIBin()
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	t.Run("List all streams in the e2e project", func(t *testing.T) {
+		cmd := exec.Command(cliPath,
+			"streams",
+			"instance",
+			"list",
+			"-o=json",
+		)
+
+		cmd.Env = os.Environ()
+		resp, err := cmd.CombinedOutput()
+		a := assert.New(t)
+		if a.NoError(err, string(resp)) {
+			var instances atlasv2.PaginatedApiStreamsTenant
+			err := json.Unmarshal(resp, &instances)
+			a.NoError(err)
+			// We expect a default instance always
+			a.Len(instances.Results, 1)
+			a.Equal(*instances.Results[0].Name, defaultInstanceName)
+		}
+	})
+
+	t.Run("Creating a streams instance", func(t *testing.T) {
+		cmd := exec.Command(cliPath,
+			"streams",
+			"instance",
+			"create",
+			"-p",
+			"AWS",
+			"-r",
+			"VIRGINIA_USA",
+			instanceName,
+			"-o=json",
+		)
```

Comment:
seems this deals with creating things and you have hardcoded the name you need to consider multiple devs running this test in parallel and the potential of name collisions, we tend to prefer creating temporary projects for this reason, please scope any creation operation to a temp project, see https://github.com/mongodb/mongodb-atlas-cli/blob/a91b145756538c4fb499dc4db71cc4a183ee7b79/test/e2e/atlas/clusters_m0_test.go#L36C1-L37

Reply:
One of the issues with creating temp projects is I believe there's currently no way to trigger a feature flag change for a created project, right?

 Streams is currently behind a feature flag while it's in private preview - none of these tests will pass with a generated project without that change.

Reply:
I've adjusted this to generate instance & connection names, but I think we might need to just use the test project because of the feature flag issues I raised.

Resolved in [2e564cc](https://github.com/mongodb/mongodb-atlas-cli/pull/2138/commits/2e564cc7202a35bd1e069298c54d1d0ed5da6f66)